File: aoc-2023/src/test/java/com/capital7software/aoc/aoc2023aoc/days/Day04Test.java
package com.capital7software.aoc.aoc2023aoc.days;

import static org.junit.jupiter.api.Assertions.assertEquals;

import com.capital7software.aoc.lib.AdventOfCodeTestBase;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

class Day04Test extends AdventOfCodeTestBase {
  private static final Logger LOGGER = LoggerFactory.getLogger(Day04Test.class);

  @BeforeEach
  void setUp() {
    var instance = new Day04();
    setupFromFile(instance.getDefaultInputFilename());
  }

  @Test
  void getSumOfAllCardPoints() {
    var instance = new Day04();
    int expected = 13;
    var actual = instance.getSumOfAllCardPoints(lines);
    assertEquals(expected, actual, "The sum of all ScratchCard points "
        + "is not what was expected: " + expected);
  }

  @Test
  void getTotalNumberOfCards() {
    var instance = new Day04();
    int expected = 30;
    var actual = instance.getTotalNumberOfCards(lines);
    assertEquals(expected, actual, "The total number of ScratchCards "
        + "is not what was expected: " + expected);
  }

  @Override
  protected Logger getLogger() {
    return LOGGER;
  }
}

File: aoc-2018/src/test/kotlin/com/capital7software/aoc/aoc2018aoc/days/Day04Test.kt
package com.capital7software.aoc.aoc2018aoc.days

import com.capital7software.aoc.lib.AdventOfCodeTestBase
import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test
import org.slf4j.Logger
import org.slf4j.LoggerFactory

class Day04Test : AdventOfCodeTestBase() {
  companion object {
    val log: Logger = LoggerFactory.getLogger(Day04Test::class.java)
  }

  override fun getLogger(): Logger {
    return log
  }

  @BeforeEach
  fun setup() {
    val instance = Day04()
    setupFromFile(instance.defaultInputFilename)
  }

  @Test
  fun testGetGuardChecksum() {
    val instance = Day04()
    val expected = 240
    val actual = instance.getGuardChecksum(lines)
    Assertions.assertEquals(expected, actual, "The checksum  "
        + "is not what was expected: $expected")
  }

  @Test
  fun testGetGuardAlternateChecksum() {
    val instance = Day04()
    val expected = 4455
    val actual = instance.getGuardAlternateChecksum(lines)
    Assertions.assertEquals(expected, actual, "The checksum "
        + "is not what was expected: $expected")
  }
}
